# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/GtkCompositorWidget.cpp
# Commit: 31f0a398817f
# Full Hash: 31f0a398817fae921abb22b709b5f8328e148aba
# Author: stransky <stransky@redhat.com>
# Date: 2024-12-10 16:36:00
# Regressor Bug: 1934497
# File Overlap Count: 1
# Description:
#   Bug 1934497 [Wayland] Update GtkCompositorWidget for WaylandSurface r=emilio
#   
#   - Replace MozContainerSurfaceLock with WaylandSurfaceLock
#   - Temporary disable GtkCompositorWidget::GetNativeLayerRoot()
#   
# ==============================================================================

diff -r 6225ae4714a9 -r 31f0a398817f widget/gtk/GtkCompositorWidget.cpp
--- a/widget/gtk/GtkCompositorWidget.cpp	Tue Dec 10 07:12:54 2024 +0000
+++ b/widget/gtk/GtkCompositorWidget.cpp	Tue Dec 10 07:12:54 2024 +0000
@@ -150,14 +150,18 @@
 #ifdef MOZ_WAYLAND
 RefPtr<mozilla::layers::NativeLayerRoot>
 GtkCompositorWidget::GetNativeLayerRoot() {
+#  if 0
   if (gfx::gfxVars::UseWebRenderCompositor()) {
     if (!mNativeLayerRoot) {
+      LOG("GtkCompositorWidget::GetNativeLayerRoot [%p] create\n",
+          (void*)mWidget.get());
       MOZ_ASSERT(mWidget && mWidget->GetMozContainer());
-      mNativeLayerRoot = layers::NativeLayerRootWayland::CreateForMozContainer(
-          mWidget->GetMozContainer());
+      mNativeLayerRoot = NativeLayerRootWayland::Create(
+          MOZ_WL_SURFACE(mWidget->GetMozContainer()));
     }
     return mNativeLayerRoot;
   }
+#  endif
   return nullptr;
 }
 #endif
@@ -208,7 +212,7 @@
 }
 #endif
 
-UniquePtr<MozContainerSurfaceLock> GtkCompositorWidget::LockSurface() {
+UniquePtr<WaylandSurfaceLock> GtkCompositorWidget::LockSurface() {
   return mWidget ? mWidget->LockSurface() : nullptr;
 }
 